js_modules/dagit/packages/ui/src/components/StyledButton.tsx

```diff
@@ -4,17 +4,34 @@ import {IconWrapper} from './Icon';
 import {SpinnerWrapper} from './Spinner';
 import {FontFamily} from './styles';
 
+export type ButtonJoinState = 'left' | 'right' | 'both';
+
 interface StyledButtonProps {
   $fillColor: string;
   $strokeColor: string;
   $textColor: string;
+  $joined?: ButtonJoinState;
 }
 
 export const StyledButton = styled.button<StyledButtonProps>`
   align-items: center;
   background-color: ${({$fillColor}) => $fillColor || 'transparent'};
   border: none;
   border-radius: 8px;
+
+  ${({$joined}) =>
+    $joined
+      ? `
+  ${$joined !== 'right' ? 'border-top-left-radius: 0;' : ''}
+  ${$joined !== 'right' ? 'border-bottom-left-radius: 0;' : ''}
+  ${$joined !== 'left' ? 'border-top-right-radius: 0;' : ''}
+  ${$joined !== 'left' ? 'border-bottom-right-radius: 0;' : ''}
+  border-left: ${$joined !== 'right' ? `1px solid rgba(255,255,255,0.2)` : 'transparent'};
```

Comment:
This is to give spacing between each button, right? Can it be `margin` instead of `border`?